aospy/test/test_data_loader.py

```diff
@@ -0,0 +1,392 @@
+        self.assertEqual(result, expected)
+
+    def test_data_name_gfdl_monthly(self):
+        # Test 'ts' and 'inst' data
+        for data_type in ['ts', 'inst']:
+            expected = 'atmos.200601-201112.temp.nc'
+            result = io.data_name_gfdl('temp', 'atmos', data_type,
+                                       'monthly', 2010, 'jja', 2000, 6)
+            self.assertEqual(result, expected)
+
+        # Test 'av' data
+        for intvl_type in ['monthly', 'mon']:
+            expected = 'atmos.2010.jja.nc'
+            result = io.data_name_gfdl('temp', 'atmos', 'av',
+                                       intvl_type, 2010, 'jja', 2000, 1)
+            self.assertEqual(result, expected)
+
+            expected = 'atmos.2006-2011.jja.nc'
+            result = io.data_name_gfdl('temp', 'atmos', 'av',
+                                       intvl_type, 2010, 'jja', 2000, 6)
+            self.assertEqual(result, expected)
+
+        # Test 'av_ts' data
+        expected = 'atmos.2006-2011.01-12.nc'
+        result = io.data_name_gfdl('temp', 'atmos', 'av_ts',
+                                   'monthly', 2010, 'jja', 2000, 6)
+        self.assertEqual(result, expected)
+
+    def test_data_name_gfdl_daily(self):
+        # Test 'ts' and 'inst' data
+        for data_type in ['ts', 'inst']:
+            expected = 'atmos.20060101-20111231.temp.nc'
+            result = io.data_name_gfdl('temp', 'atmos', data_type,
+                                       'daily', 2010, None, 2000, 6)
+            self.assertEqual(result, expected)
+
+        # Test 'av' data; this should raise an error
+        with self.assertRaises(NameError):
+            io.data_name_gfdl('temp', 'atmos', 'av',
+                              'daily', 2010, None, 2000, 6)
+
+        # Test 'av_ts' data
+        expected = 'atmos.2006-2011.01-12.nc'
+        result = io.data_name_gfdl('temp', 'atmos', 'av_ts',
+                                   'daily', 2010, None, 2000, 6)
+        self.assertEqual(result, expected)
+
+    def test_data_name_gfdl_hr(self):
+        # Test 'ts' and 'inst' data
+        for data_type in ['ts', 'inst']:
+            expected = 'atmos.2006010100-2011123123.temp.nc'
+            result = io.data_name_gfdl('temp', 'atmos', data_type,
+                                       '3hr', 2010, None, 2000, 6)
+            self.assertEqual(result, expected)
+
+        # Test 'av' data; this should raise an error
+        with self.assertRaises(NameError):
+            io.data_name_gfdl('temp', 'atmos', 'av',
+                              '3hr', 2010, None, 2000, 6)
+
+        # Test 'av_ts' data
+        expected = 'atmos.2006-2011.01-12.nc'
+        result = io.data_name_gfdl('temp', 'atmos', 'av_ts',
+                                   '3hr', 2010, None, 2000, 6)
+        self.assertEqual(result, expected)
+
+    @unittest.expectedFailure  # This aspect of this function is obselete
```

Comment:
I should have been more descriptive here.  Within ```utils.io.data_name_gfdl``` it is stated:
```python
elif intvl_type in ['seasonal', 'seas']:
      # 2015-04-29: This function is obsolete.  Should fix this
      # eventually.  But I almost never use seasonal data, so not a
      # priority.
      # label, val = _intvl_indices_and_label(intvl, intvl_type)
      label = label.upper()
```
The way the function is written currently, you'll get a ```NameError``` whenever this ```elif``` statement is active, because ```label``` will not be defined before the ```label = label.upper()``` line.  As far as I can tell the function used in the call that defines ```label```, which was commented out here, has been removed from aospy at some point in the past.  Should we just delete this part of the function / tests of it for now?